federation/pkg/federation-controller/service/service_controller.go

```diff
@@ -731,6 +748,17 @@ func (slice ingress) Swap(i, j int) {
 // getClusterServiceIngresses returns cluster service ingresses for given region, zone and cluster name
 func getClusterServiceIngresses(ingress *fed.FederatedServiceIngress,
 	region, zone, cluster string) *fed.ClusterServiceIngress {
+	if ingress.Endpoints != nil && ingress.Endpoints[region] != nil && ingress.Endpoints[region][zone] != nil &&
+		ingress.Endpoints[region][zone][cluster] != nil {
+		return ingress.Endpoints[region][zone][cluster]
+	}
+	return nil
+}
+
+// getOrCreateClusterServiceIngresses returns cluster service ingresses for given region, zone and cluster name if
+// exist otherwise creates one and returns
+func getOrCreateClusterServiceIngresses(ingress *fed.FederatedServiceIngress,
```

Comment:
This does not really create anything.  It only replaces nil maps with empty maps, and returns an empty ClusterServiceIngress rather than a nil?  This logic is confusing.